StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradePotionOfStrength.cpp

```diff
@@ -0,0 +1,66 @@
+#include "UpgradePotionOfStrength.h"
+#include "EngineMath.h"
+#include "ECS/Entity.h"
+
+UpgradePotionOfStrength::UpgradePotionOfStrength(Entity* parent) : Upgrade(parent)
+{
+	// Set the name of the upgrade
+	SetName("PotionOfStrength");
+
+	// Set type
+	SetType(F_UpgradeType::PLAYER);
+
+	m_StartingPrice = 100;
+	m_Price = m_StartingPrice;
+	m_IsActive = false;
+
+	m_Description = "Temporarily upgrades the attack damage of both ranged and melee. Boost is removed at the end of the level.";
+	m_ImageName = "PotionOfStrength.png";
+}
+
+UpgradePotionOfStrength::~UpgradePotionOfStrength()
+{
+}
+
+void UpgradePotionOfStrength::IncreaseLevel()
+{
+	m_Price = (++m_Level + 2) * 100;
+	ApplyStat();
+}
+
+void UpgradePotionOfStrength::ApplyStat()
```

Comment:
If you want it to take effect the first time you buy it then you have to use the function ApplyBoughtUpgrade() and call on ApplyStat() in there. This should probably be said in the tutorial my bad.

Reply:
Now that I look through the shop and upgradeManager code it seems like the increase in level for UpgradeManager is actually applying your damage change. This is a bug that I need to fix since it clashes, in other upgrades, with ApplyBoughtUpgrade(). I need to take a closer look at this.

Reply:
This bug will be fixed in my UpgradeGUI pullrequest. So add the ApplyBoughtUpgrade() and call ApplyStat() in it.